examples/scheduler/src/main/kotlin/cnt/scheduler/AnnotationBasedScheduler.kt

```diff
@@ -0,0 +1,31 @@
+package cnt.scheduler
+
+import io.quarkus.scheduler.Scheduled
+import org.eclipse.microprofile.config.inject.ConfigProperty
+import org.slf4j.Logger
+import javax.enterprise.context.ApplicationScoped
+
+@ApplicationScoped
+class AnnotationBasedScheduler(
+    private val sendBookNewsletter: SendBookNewsletter,
+    @ConfigProperty(name = "scheduler.book-newsletter.enabled") private val sendBookNewsletterEnabled: Boolean
+) {
+
+    @Scheduled(identity = "NewsletterScheduler", every = "{scheduler.book-newsletter.interval}")
+    fun newsletterScheduler() {
+        if (sendBookNewsletterEnabled) {
+            sendBookNewsletter()
+        }
+    }
+}
+
+@ApplicationScoped
+class SendBookNewsletter(
+    private val logger: Logger = getLogger<SendBookNewsletter>()
```

Comment:
Does the logger need to be a constructor parameter?

Reply:
I would say yes I it isolates logic which is not relevant for this scheduler.
It keeps the relevant business logic cleaner.

Reply:
While I would say you are generally right to inject anything that is an "outside" / "out of scope" component. In this case it is more typical to have the logger hard-wired as part of the class.

Reasons why I would prefer that:
1. Logger is not a bean that is available in the application context, so I assume in this case the application uses the default value as a fallback.
2. Loggers have a 1:1 relationship to their owning class, so the only reason to make them exchangeable would be to be able to mock them. Mocking a logger adds unnecessary effort to testing, because now you would need to provide behaviour to it. If one wants to verify log output, it is a bit easier to use something like the LogRecorder.
3. If you would provide a `Logger` bean in the application context, all other beans with a `Logger` in their constructor would share the same logger. I assume that would not be the desired outcome.